### ISSUE TITLE
raidboss: add P1S shining cells combo light/fire + left/right + in/out calls

### ISSUE THREAD

**Issue by quisquous** [quisquous/cactbot#4081]
This combo mechanic doesn't have a trigger yet.

**Comment by quisquous** [comment]
# Aetherflail cast+ability -> second ability

During Shining Cells there's an initial Aetherflail cast, followed by a second Aetherflail ability.  Here are the initial casts, followed by possible second casts.  Am I missing combos here? It's suspicious that 6606 and 6607 only have one followup option.

6602 -> 660B,660D
6603 -> 660A,660C
6606 -> 6609
6607 -> 6608

Find initial casts
```
$ grep -h "^20.*Aetherflail" /d/ffxivlogs/log/Network*2022*.log | sed "s/^.\{59\}//" | sed "s/|.*//" | sort | uniq
```

Find any casts or abilities
```
$ grep -h "^2.*Aetherflail" /d/ffxivlogs/log/Network*2022*.log | sed "s/^.\{59\}//" | sed "s/|.*//" | sort | uniq
```

ACT Regex for imported logs

`(gaoler's flail| 14.*Intemperate Torment| 14.*Shining Cells|Powerful (Fire|Light)|aetherflail|StatusAdd.*(E0000000|Erich).*(E0000000|Erich))`

# Aetherflail abilities from xivapi

Looking at all possibilities for ability ids, there's numbers that aren't used? The 65DF-65E4 block doesn't appear for me anywhere in my logs.  Additionally, 6604 and 6605 are holes in the 6602-660D block that are otherwise used which seems suspicious.

https://xivapi.com/search?indexes=action&string=Aetherflail&columns=Name,ID&pretty=true

# Initial Gaoler's Flail fake cast (mech): first damage -> second damage

Gaoler's Flail has a number of fake cast and ability ids, e.g. 6F56-6601.  It also has a number of real cast and ability damage ids (casted in shining cells, uncasted elsewhere), e.g. 6DA6-6DAD.  Here's all the initial GF fake casts into the damage ids they represent.

65F6 (l->r): 6DA6 -> 6DAB
65F7 (r->l): 6DA7 -> 6DAA
65F8 (out->in): 6DA8 -> 6DAD
65F9 (out->in): 6DA8 -> 6DAD
65FA (in->out): 6DA9 -> 6DAC
65FB (in->out): 6DA9 -> 6DAC

Therefore:

6DA6: left (first)
6DA7: left (first)
6DA8: out (first)
6DA9: in (first)
6DAA: right (second)
6DAB: right (second)
6DAC: out (second)
6DAD: in (second)

# Aetherflail ability1 -> ability2: damage1 -> damage2

During shining cells there's two aetherflail abilities and two gaoler's flail damage abilities, where the timing for these is roughly ~2s apart.

During shining cells, we can also look at what damage abilities goes with the pair of Aetherflail casts.  Am I missing two pairs where 6DA8 (out) comes first?

6602 -> 660B: 6DA6 -> 6DAC (left -> out)
6602 -> 660D: 6DA6 -> 6DAD (left -> in)
6603 -> 660A: 6DA7 -> 6DAC (right -> out)
6603 -> 660C: 6DA7 -> 6DAD (right -> in)
6606 -> 6609: 6DA9 -> 6DAB (in -> right)
6607 -> 6608: 6DA9 -> 6DAA (in -> left)

# Conclusion so far?

Surely there's some animation data being sent in network data, but from pure casts it doesn't look like you can predict the second aetherflail from the first.

**Comment by quisquous** [comment]
Arguably, we could call out "In + Fire + ???" or "In + Fire + Left/Right" or alternatively just add a second "In/Out/Left/Right" call simultanously with the "Fire/Light" call but that's noisy and incomplete too.

**Comment by Legends0** [comment]
I had looked at this and could not find a way to predict what the second aetherflail would be. I do not know how the client gets that information other than we see animation that did not seem to be tracked in log file. Because of this missing information, a trigger has largely been left out.

**Comment by panicstevenson** [comment]
> Arguably, we could call out "In + Fire + ???" or "In + Fire + Left/Right" or alternatively just add a second "In/Out/Left/Right" call simultanously with the "Fire/Light" call but that's noisy and incomplete too.

Late reply here, but I feel like one of these always has to be in and the other one always has to be out, just by anecdotal evidence that I call this out for my group and I call it out before the animation completes and I don't think I've been wrong yet. It doesn't help that we only need to run it once a week though and human brains are very fallible.

**Comment by quisquous** [comment]
I agree with that, but I don't know that's enough information to call it in all cases too, just more information for the second round.  I'm not sure that the L/R alternates either (but maybe it does?)

**Comment by quisquous** [comment]
Copying @panicstevenson's hard work here.  The final shining cells has a in/out+left/right x2 followed by an in/out combo.:

6604 -> 660D: 6DA8 -> 6DAD (out -> in) [link](https://www.fflogs.com/reports/f8QnWVYFbgvHPJcT#fight=9&type=casts&view=events&hostility=1&start=5020364&end=5107485)
6605 -> 660C: 6DA8 -> 6DAD (out -> in) [link](https://www.fflogs.com/reports/2zQ9XvYpkrm3fnda#fight=7&view=events&hostility=1&start=4104955&end=4280123&type=casts)
6606 -> 660B: 6DA9 -> 6DAC (in -> out) [link](https://www.fflogs.com/reports/hdQX8DtAPx1RmFVb#fight=last&type=casts&hostility=1&start=2241274&end=2354004&view=events)
6607 -> 660A: 6DA9 -> 6DAC (in->out) [link](https://www.fflogs.com/reports/pChWjNPx3KkRVJH8#fight=8&view=events&start=4010860&end=4081064&hostility=1&type=casts)

I think this means that we are missing triggers for these in/outs, however we can't call 6606/6608 in the same way we call 65F8 in a trigger because 6606 can be in->out but also in->right.

**Comment by panicstevenson** [comment]
> I agree with that, but I don't know that's enough information to call it in all cases too, just more information for the second round. I'm not sure that the L/R alternates either (but maybe it does?)

Also to update, we found this isn't true; it can be Right/In followed by Right/Out (or Left, etc.).
Worth noting that Out can never be first when paired with a direction, so it can never be Out -> Right or Out -> Left. It can only ever be Out (dropped on left or right side) -> In (dropped on other side).

**Comment by Loskh** [comment]
The combo is determined by Erichthonios's weapon ,and his weapon is determined by ActorControl.
ActorControl directs mobs behaviors ,but  ACT Plugin only processes few catagories.[ActorControlType in Sapphire](https://github.com/SapphireServer/Sapphire/blob/aef56c9f336473472147cdeabc0cbc97f440f023/src/common/Network/CommonActorControl.h#L16)
According to disassemble result , I think that a actorcontrol packet with a ActorControlType 0x1E directs something related to  BNpcParts, for example, Erichthonios's weapon .
[P1.log with extra ActorControlType](https://github.com/quisquous/cactbot/files/8523815/P1.log)
And in a8, Brawler changes its weapons with similarly method.   It directs something related to  ModelState, and its value is 0x1F.
[Brawler in A8.log](https://github.com/quisquous/cactbot/files/8523881/Brawler.in.A8.log)
Tsukuyomi changes her model by 0x3F.
The type values ranges widely. I think it should be researched detailed.
